components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,215 @@
+import pg from "pg";
+
 export default {
   type: "app",
   app: "postgresql",
-  propDefinitions: {},
+  propDefinitions: {
+    table: {
+      type: "string",
+      label: "Table",
+      description: "Database table",
+      async options() {
+        return this.getTables();
+      },
+    },
+    column: {
+      type: "string",
+      label: "Column",
+      description: "The name of a column in the table to use for deduplication. Defaults to the table's primary key",
+      async options({ table }) {
+        return this.getColumns(table);
+      },
+    },
+    query: {
+      type: "string",
+      label: "SQL Query",
+      description: "Your custom SQL Query using `$1`, `$2`... to represent values (eg. `INSERT INTO users(name, email) VALUES($1, $2)`",
+    },
+    values: {
+      type: "string[]",
+      label: "Values",
+      description: "List of values represented in your SQL Query above",
+    },
+    value: {
+      type: "string",
+      label: "Lookup Value",
+      description: "Value to search for",
+      async options({
+        table, column, prevContext,
+      }) {
+        const limit = 20;
+        const { offset = 0 } = prevContext;
+        return {
+          options: await this.getColumnValues(table, column, limit, offset),
+          context: {
+            offset: limit + offset,
+          },
+        };
+      },
+    },
+    rowValues: {
+      type: "object",
+      label: "Row Values",
+      description: "Enter the column names and respective values as key/value pairs, or with structured mode off as `{{{columnName:\"columnValue\"}}}`",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async getClient() {
+      const { Client } = pg;
+      const {
+        user,
+        password,
+        host,
+        port,
+        database,
+      } = this.$auth;
+      const connectionString = `postgresql://${user}:${password}@${host}:${port}/${database}`;
+      const client = new Client({
+        connectionString,
+      });
+      await client
+        .connect()
+        .catch((err) => console.error("Connection error", err.stack));
+      return client;
+    },
+    async endClient(client) {
+      return client.end();
+    },
+    /**
+     * Executes SQL query and returns the resulting rows
+     * @param {string} text - SQL query to execute
+     * @param {array} [values] - values to substitute into the given query
+     * @returns Array of rows returned from the given SQL query
+     */
+    async executeQuery(text, values = null) {
+      const client = await this.getClient();
+      const query = values
+        ? {
+          text,
+          values,
+        }
+        : text;
+      try {
+        const { rows } = await client.query(query);
+        return rows;
+      } finally {
+        await this.endClient(client);
+      }
+    },
```

Comment:
As a suggestion I would leave the `query` object to be configurable outside of this function to make it available to other functions. It migth be helpful in case you want to add more properties to `query`.

```suggestion
    async executeQuery(query) {
      const client = await this.getClient();
      try {
        const { rows } = await client.query(query);
        return rows;
      } finally {
        await this.endClient(client);
      }
    },
```